### ISSUE TITLE
Typst

### ISSUE THREAD

**Issue by TimeTravelPenguin** [catppuccin/catppuccin#2491]
### Link to repository

https://github.com/TimeTravelPenguin/typst.git

### Screenshots

![frappe](https://github.com/user-attachments/assets/48c293fe-f2b4-4919-8ba8-f9887a7c10d0)
![latte](https://github.com/user-attachments/assets/90cbc492-9191-43d9-bc37-70f1066c3ef7)
![macchiato](https://github.com/user-attachments/assets/9953824f-3bd0-4f93-8210-6ffc05c79b0e)
![mocha](https://github.com/user-attachments/assets/99521560-9dc3-4773-be42-18a654ff4ed7)

### Any additional comments?

This project is currently perfectly functional. I do, however, plan to finish off some minor changes to make it a bit more convenient to use. I will also be publishing the package once that step is complete, and if accepted.

There is still work to be done, which I will continue to complete over time, but that is mainly additional functionality (and may not even happen). I currently have a todo list in the readme.

### Submission Guidelines

- [X] I have read and followed Catppuccin's submission guidelines.

**Comment by uncenter** [comment]
Do the tmThemes from https://github.com/catppuccin/bat/tree/main/themes work for your case? They look similar. Probably better to reuse our existing port if possible!

**Comment by TimeTravelPenguin** [comment]
> Do the tmThemes from [catppuccin/bat@`main`/themes](https://github.com/catppuccin/bat/tree/main/themes?rgh-link-date=2024-08-04T09%3A41%3A39Z) work for your case? They look similar. Probably better to reuse our existing port if possible!

Aha! Excellent, I was not able to find an existing tool, but this looks exactly like what I was looking for. I will update my build script and readme now!

**Comment by pinkcreeper100** [comment]
may i mention the existence of [typpuccino](https://typst.app/universe/package/typpuccino)

**Comment by TimeTravelPenguin** [comment]
> may i mention the existence of [typpuccino](https://typst.app/universe/package/typpuccino)

Oh wow, I didn't see this one 😮

**Comment by TimeTravelPenguin** [comment]
Hmm, it seems that the typpuccino package doesn't deal with syntax highlighting, nor does it have any current build utils to keep the theming up to date.

I will leave it with the powers that be as to whether my repo is a good fit or if I should instead take an alternate route of action.

**Comment by sgoudham** [comment]
Hey :wave:,
Thank you for your work on this! Great job on the previews.

With regards to accepting this repository over something like typpuccino, I'd be happy to accept this into the organisation! I'm not sure if was intentional or unintentional for them to publish outwith the organisation but no one is obliged to raise an issue and get their repository merged back so I really appreciate this!

With regards to publishing to the built-in package manager, we usually publish under a shared organisation account to avoid a dependence on one single maintainer if they wish to leave the organisation later on. I presume the staff team can create an account and upload via an API key of sorts? 

For the repository itself, I've got some things:

1. [Generating every single variation of the catwalk image](https://github.com/TimeTravelPenguin/typst/blob/28bdefa5331fd65a1d3690b2af6ad9942964f6ee/justfile#L44-L60) is, imo, unnecessary and isn't carried out in other ports so I would remove the code to do that and have the assets directory look like the following screenshot (where `preview.webp` is the catwalk image in its composite form):

	![image](https://github.com/user-attachments/assets/1e7d6863-4cb4-4f8e-8d73-9918a05d0bdf)

2. Our convention for whiskers ports is for the tera file to take the filename `<port-name>.tera` so I'd appreciate if you could rename [catppuccin.tera](https://github.com/TimeTravelPenguin/typst/blob/main/catppuccin.tera) -> `typst.tera`
3. It'd be good to pin the [GitHub link](https://github.com/TimeTravelPenguin/typst/blob/28bdefa5331fd65a1d3690b2af6ad9942964f6ee/justfile#L10-L13) to a commit hash for pulling down the `.tmTheme` files to avoid potential unwanted changes every time you build.
4. You can remove the header indicating that the repository is in development and to convert the TODO list into issues on the repository to avoid cluttering the README.

Think we can merge the repository into the organisation after the above comments are addressed and we can figure out a nice automated publishing system (hopefully that's possible!)

Thanks again for your work!

**Comment by uncenter** [comment]
A few little things I noticed as well:

- In the README, I believe "Typst's" is misspelled as "typist's".
- The catppuccin.com link in the repository sidebar should be removed (though in the future you can link to something like the package page or a documentation page).
- In the Whiskers template, you can get rid of [the `app` and `author` fields](https://github.com/TimeTravelPenguin/typst/blob/28bdefa5331fd65a1d3690b2af6ad9942964f6ee/catppuccin.tera#L7-L11) - those are just for example in our Whiskers repository, you don't actually need it :)

**Comment by TimeTravelPenguin** [comment]
@sgoudham Thank you for your feedback. I haven't fully familiarised myself with the [publishing](https://github.com/typst/packages?tab=readme-ov-file#published-packages) process, but it has been on my list of things to do. For the current time, it appears that they accept packages via a PR on the repo in the previous link.

I have applied most of your mentioned points. The only one I have yet to do is apply your third point. I am not sure how to go about this, so any additional info on how to go about this would be awesome!

@uncenter Thanks for pointing these out! I have been working in between my university work, so there were bound to be a few typos!

I have pushed my changes to main and rebased my dev branch. With the exception of the one task mentioned above, I believe everything is ready (except moving the todo list to issues -- I will sit down later and write each one with more info).

Lastly, should I leave the links to the shields.io badge and image preview within the readme as links to my repo? Or am I right to assume they will be changed after the transfer?

Thanks! I look forward to continuing my work on this. My eyes are grateful :)

**Comment by sgoudham** [comment]
> For the current time, it appears that they accept packages via a PR on the repo in the previous link.

I had a look at the repository and I'm a little bit annoyed that the package's source files are duplicated upstream as it's a manual effort to have someone raise the PR to update the files on fixes/improvements. Not to mention that all the files are duplicated somewhat unnecessarily in my opinion. In any case, we can deal with publishing after transfer, thanks for the link.

> I have applied most of your mentioned points. The only one I have yet to do is apply your third point. I am not sure how to go about this, so any additional info on how to go about this would be awesome!

You can take the commit hash of a commit and reference any file on GitHub according to that hash, ensuring that the same file at that point in time is retrieved.

In your case, I'd take the commit hash of the latest commit on catppuccin/bat and include it in the URL like so

```diff
- wget -q https://github.com/catppuccin/bat/raw/main/themes/Catppuccin%20Latte.tmTheme -O tmThemes/latte.tmTheme &
- wget -q https://github.com/catppuccin/bat/raw/main/themes/Catppuccin%20Frappe.tmTheme -O tmThemes/frappe.tmTheme &
- wget -q https://github.com/catppuccin/bat/raw/main/themes/Catppuccin%20Macchiato.tmTheme -O tmThemes/macchiato.tmTheme &
- wget -q https://github.com/catppuccin/bat/raw/main/themes/Catppuccin%20Mocha.tmTheme -O tmThemes/mocha.tmTheme &
+ wget -q https://github.com/catppuccin/bat/raw/d3feec47b16a8e99eabb34cdfbaa115541d374fc/themes/Catppuccin%20Latte.tmTheme -O tmThemes/latte.tmTheme &
+ wget -q https://github.com/catppuccin/bat/raw/d3feec47b16a8e99eabb34cdfbaa115541d374fc/themes/Catppuccin%20Frappe.tmTheme -O tmThemes/frappe.tmTheme &
+ wget -q https://github.com/catppuccin/bat/raw/d3feec47b16a8e99eabb34cdfbaa115541d374fc/themes/Catppuccin%20Macchiato.tmTheme -O tmThemes/macchiato.tmTheme &
+ wget -q https://github.com/catppuccin/bat/raw/d3feec47b16a8e99eabb34cdfbaa115541d374fc/themes/Catppuccin%20Mocha.tmTheme -O tmThemes/mocha.tmTheme &
```

> (except moving the todo list to issues -- I will sit down later and write each one with more info)

I saw that you added issue templates into the codebase, it'd be great if you could use the newer [issue forms syntax ](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms)for specifying the templates. There are some previous examples in repositories like [catppuccin/tmux](https://github.com/catppuccin/tmux/tree/main/.github/ISSUE_TEMPLATE) and [catppuccin/jetbrains](https://github.com/catppuccin/jetbrains/tree/main/.github/ISSUE_TEMPLATE).

> Lastly, should I leave the links to the shields.io badge and image preview within the readme as links to my repo? Or am I right to assume they will be changed after the transfer?

It'd be preferred if you can update them before transfer if you'd like to but most people raise PRs to update them after transfer. I'm not particularly fussed about it!

**Comment by TimeTravelPenguin** [comment]
> I had a look at the repository and I'm a little bit annoyed that the package's source files are duplicated upstream as it's a manual effort to have someone raise the PR to update the files on fixes/improvements. Not to mention that all the files are duplicated somewhat unnecessarily in my opinion. In any case, we can deal with publishing after transfer, thanks for the link.

Yeah, I agree. I am really hoping that devs can eventually provide a way to pull packages in a better way. Be it through a better service or directly from GitHub. However, I can understand the difficulties and security concerns.

> You can take the commit hash of a commit and reference any file on GitHub according to that hash, ensuring that the same file at that point in time is retrieved.

Aha! I see what you mean now. I misunderstood what you meant. This is no issue. When I have more time I can add a check in the Just file that prints a warning if the files are outdated.

> I saw that you added issue templates into the codebase, it'd be great if you could use the newer [issue forms syntax ](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms)for specifying the templates.

Thanks for the link! I was going to mention the addition of the templates but it seems you beat me to it! I didn't know there were new template formats, so that is interesting to see! I will work on that after this.

> It'd be preferred if you can update them before transfer if you'd like to but most people raise PRs to update them after transfer. I'm not particularly fussed about it!

Done! Not a big deal if it looks strange for the moment. Extra motivation to get through it all asap!

**Comment by TimeTravelPenguin** [comment]
Issue templates have been updated @sgoudham. I have used links with the Catppuccin URL so that they will work once transferred. I will go over them once more afterwards to double-check when the URLs work!

**Comment by TimeTravelPenguin** [comment]
Thanks @sgoudham! I have just completed the transfer, now.

The only reason I have not used `.webp` images is due to Typst not currently supporting the export option. Since I tried to make sure the build process was as inclusive as possible of all the different parts, I decided to leave it as is until I had a way to convert them without doing so manually. Of course, if they need to be converted manually, then so be it!

**Comment by TimeTravelPenguin** [comment]
> My only nitpicks are that the URLs in the README should be [relative](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#relative-links) and that all assets should be in the `.webp` format.

@sgoudham Were you referring to [these](https://github.com/catppuccin/typst/blob/d3e9d948eb388dbd979d9983c1a957e4584ba47d/README.md?plain=1#L14-L35) links in particular?

If not, which? I'm not certain as to which of the remaining links you are referring to.

Also, what would you suggest to be the easiest way to convert `.png` to `.webp`? I just want to make future work a bit more streamlined :)

**Comment by sgoudham** [comment]
> @sgoudham Were you referring to [these](https://github.com/catppuccin/typst/blob/d3e9d948eb388dbd979d9983c1a957e4584ba47d/README.md?plain=1#L14-L35) links in particular?

Yup, those were the links I was referring to!

> Also, what would you suggest to be the easiest way to convert .png to .webp? I just want to make future work a bit more streamlined :)

I personally use [Imagemagick]() which is a standard utility with the following one liner but I know technically [cwebp](https://developers.google.com/speed/webp/docs/cwebp) is the official tool.

```shell
for i in latte frappe macchiato mocha; magick -quality 90 $i.webp $i.png; end
```

Just one more thing, it would be great if you could use conventional commits in your repository as outlined in the recommendations section of the [CONTRIBUTING.md](https://github.com/catppuccin/catppuccin/blob/main/CONTRIBUTING.md#recommendations)

**Comment by TimeTravelPenguin** [comment]
> I personally use [Imagemagick](https://imagemagick.org/script/convert.php)

Awesome, thanks. I am familiar with imagemagick. I just always forget it exists 😂

> Just one more thing, it would be great if you could use conventional commits in your repository as outlined in the recommendations section of the [CONTRIBUTING.md](https://github.com/catppuccin/catppuccin/blob/main/CONTRIBUTING.md?rgh-link-date=2024-08-13T17%3A28%3A26Z#recommendations)

Ah! I have been looking for this. I have been wanting to look into learning exactly this but had never known where to look. I will certainly be adapting this. Thanks!
